dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClusterCPUChartTaskTest.java

```diff
@@ -173,10 +163,70 @@ public void chartWithEmptyDateRange_throwsException() throws Exception {
         "The chart has to include at least one day. Received 0 days.", exception.getMessage());
   }
 
+  @Test
+  public void doRun_clouderaReturns4xx_throwsCriticalException() throws Exception {
+    // GIVEN: There is a valid cluster
+    initClusters(ClouderaClusterDTO.create("id1", "first-cluster"));
+    String firstClusterServicesJson = servicesJson;
+    mockHttpRequestToFetchClusterCPUChart(
```

Comment:
No, it must be tested with WireMock server, we can add some timeouts to responses as well